### PR TITLE
chonburi-font: init at unstable-2021-09-15

### PR DESCRIPTION
**Pull request by toastal**
OFL font: https://github.com/cadsondemak/chonburi

###### Motivation for this change

Another free font, like `kanit-font`

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 138131` at f03f1068 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>chonburi-font</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/data/fonts/chonburi/default.nix:14:3:
```
   |
14 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/data/fonts/kanit/default.nix [hunk 1]
```diff
@@ -12,12 +12,16 @@ stdenv.mkDerivation rec {
...
+    runHook preInstall
+
     mkdir -p $out/share/doc/${pname}/css/ $out/share/fonts/{opentype,truetype}
 
     cp $src/OFL.txt $src/documentation/{BRIEF.md,features.html} $out/share/doc/${pname}
     cp $src/documentation/css/fonts.css $out/share/doc/${pname}/css
     cp $src/fonts/otf/*.otf $out/share/fonts/opentype
     cp $src/fonts/ttf/*.ttf $out/share/fonts/truetype
+
+    runHook postInstall
```
Comment (figsoda) [inline pkgs/data/fonts/kanit/default.nix:24]: seems unrelated, maybe this should be in a separate commit or a separate pull request?

File: pkgs/data/fonts/chonburi/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    owner = "cadsondemak";
+    repo = pname;
+    rev = "daf26bf77d82fba50eaa3aa3fad905cb9f6b5e28";
+    sha256 = "sha256-oC7ZCfNOyvGtqT9+Ap/CfCHzdWNzeCuac2dJ9fctgB8=";
+  };
+
+  installPhase = ''
+    mkdir -p $out/share/doc/${pname} $out/share/fonts/{opentype,truetype}
+
+    cp $src/OFL.txt $src/BRIEF.md $out/share/doc/${pname}
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/chonburi/default.nix:17]:
```suggestion
    mkdir -p $out/share/doc/chonburi $out/share/fonts/{opentype,truetype}

    cp $src/OFL.txt $src/BRIEF.md $out/share/doc/chonburi
```

File: pkgs/data/fonts/kanit/default.nix (hunk 1 above)
Comment (toastal) [reply pkgs/data/fonts/kanit/default.nix:24]: Ouch. Edited the wrong file. I'm maintaining this file too so I was using it as reference. Since this is the case, I do think I'd prefer they stay similar; I will do as a separate commit.

**Comment by figsoda** [comment]
Result of `nixpkgs-review pr 138131` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>chonburi-font</li>
    <li>kanit-font</li>
  </ul>
</details>

**Review by figsoda — APPROVED** [review]
the second commit message would probably be better as `kanit-font: [...]`, otherwise lgtm
